src/plugins/NEMD/MettDeamon.h

```diff
@@ -422,15 +418,15 @@ class Reservoir
 	std::unique_ptr<BinQueue> _binQueue;
 	uint64_t _numMoleculesRead;
 	uint64_t _nMaxMoleculeID;
-	uint32_t _nMoleculeFormat;
-	uint8_t _nReadMethod;
+	MoleculeFormat _nMoleculeFormat;
+	ReadReservoirMethods _nReadMethod;
 	double _dReadWidthY;
 	double _dBinWidthInit;
 	double _dBinWidth;
 	double _dInsPercent;  // only insert percentage of reservoir density
 	std::vector<Molecule> _particleVector;
 	std::vector<uint32_t> _vecChangeCompIDs;
-	std::vector<DensityStruct> _density;
+	double _density; // Global density of reservoir
```

Comment:
This vector was supposed to hold the densities and numbers of particles in the reservoir separated by component id. Reasons to remove it are:

1. First and foremost because it was set but never used or documented how it could be used in the future.
2. It was probably not even set properly in all cases which was a result of the different read methods which handled the particles differently.